File: Q8_ReverseLinkedList.c
#include <stdio.h>
#include <stdlib.h>
struct Node{int data;struct Node* next;};
struct Node* reverse(struct Node* head){struct Node* prev=NULL;struct Node* cur=head;while(cur){struct Node* nxt=cur->next;cur->next=prev;prev=cur;cur=nxt;}return prev;}
void print(struct Node* h){while(h){printf("%d ",h->data);h=h->next;}}
int main(){struct Node* head=malloc(sizeof(struct Node));head->data=1;head->next=malloc(sizeof(struct Node));head->next->data=2;head->next->next=NULL;head=reverse(head);print(head);}

File: Q8_ReverseLinkedList.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
def reverse(head):
    prev=None
    cur=head
    while cur:
        nxt=cur.next
        cur.next=prev
        prev=cur
        cur=nxt
    return prev
def print_list(h):
    while h:
        print(h.data,end=" ")
        h=h.next
head=Node(1)
head.next=Node(2)
head=reverse(head)
print_list(head)